cmd/juju/cloud/add.go

```diff
@@ -109,6 +110,9 @@ func (c *AddCloudCommand) Init(args []string) (err error) {
 		c.Cloud = args[0]
 	}
 	if len(args) > 1 {
+		if c.CloudFile != args[1] && c.CloudFile != "" {
```

Comment:
this is a little confusing to me.  why are we checking cloudfile == args[1] ? 

It seems like this would allow you to enter `juju add-cloud maas -f foo.yaml foo.yaml`  which I don't think we should allow.  IMO the flag and positional should be mutually exclusive.

seems like we should just check
```go
if len(args) > 1 {
    if c.CloudFile != "" {
        // return error
    }
    c.CloudFile = args[1]
}
```